### ISSUE TITLE
Datapacks per worlds

### ISSUE THREAD

**Issue by CodeDoctorDE** [bergerhealer/MyWorlds#18]
Datapacks will only run in the worlds which are included. 
When a datapack uses @a, it will only use the player which are in the world!

**Comment by weeryan17** [comment]
Maybe as an config option, but vannila behavior @a affects everyone in
every world. So this shouldn't be default.

**Comment by CodeDoctorDE** [comment]
Yes that's my request. because many datapacks uses this operators and it's the best thing to use it in only this world. i don't know if you can add an option for the execute command

**Comment by weeryan17** [comment]
Actually thinking about it well it is technically possible it would be
really hard to implement as you would have to override vannila behavior.

**Comment by LaurenceBarnes** [comment]
It would make a lot more sense to have multiple servers for this, check out a proxy, such as https://velocitypowered.com/

**Comment by CodeDoctorDE** [comment]
The problem is that you can't create a data pack only server with multiple worlds

**Comment by LaurenceBarnes** [comment]
you should avoid datapacks anyway. As long as they do not have "functions" and just do stuff with worldgen or recipes, it is okay, but if they do have functions, they perform a lot worse than bukkit plugins.
Datapacks are simply supposed to work as they are, if you want stuff per-world, use/write plugins.

**Comment by weeryan17** [comment]
> you should avoid datapacks anyway. As long as they do not have "functions" and just do stuff with worldgen or recipes, it is okay, but if they do have functions, they perform a lot worse than bukkit plugins.
> Datapacks are simply supposed to work as they are, if you want stuff per-world, use/write plugins.

What? Ok first off their are a lot of things that exist in datapack form that don't exist in plugin form. Not everyone knows how to write plugins, so if they can find something, even if it is a datapack that does something they want it to do they should be fully allowed to do so.
So plugins are not a replacement for datapacks.

And this is a valid suggestion, as mentioned above in this issue shouldn't be the default, but people do use separate worlds as completely separate things, it is in fact quite common to have a survival world, and a creative world.

**Comment by LaurenceBarnes** [comment]
if you want to separate stuff safely, have two servers and a proxy so people can switch between them. It is generally not possible to have datapacks per-world.
And what does a datapack do that plugins cannot do? The BukkitAPI is really simple, and plugins are the only sane replacement for any datapack using functions (if you do not know what they are, https://minecraft.fandom.com/wiki/Tutorials/Creating_a_data_pack#Functions)

**Comment by weeryan17** [comment]
> if you want to separate stuff safely, have two servers and a proxy so people can switch between them. It is generally not possible to have datapacks per-world.

Not everybody has access to multiple servers, and even then multiple servers depending on how many people you have is not worth it. Now being not possible? maybe. I'll leave that up to berger to find out.

> And what does a datapack do that plugins cannot do? The BukkitAPI is really simple, and plugins are the only sane replacement for any datapack using functions (if you do not know what they are, https://minecraft.fandom.com/wiki/Tutorials/Creating_a_data_pack#Functions)

Not everyone can code, just flat out, and even those who can code it still would be easier to use datapacks in a lot of scenarios. Yes plguins can do a lot more then datapacks, but that's not what this is about. And again not everyone can code (or learn to code before you bring that up).

**Comment by CodeDoctorDE** [comment]
Additionally datapacks are lightweight, easy to understand and easy to share. If you want to have a minigames server without plugins, it will be the best to have such a functionality to send players in different worlds with different behaviours. For example you can teleport the players from the Lobby world to the minigames world if the game starts.
Currently datapacks cannot be used with a proxy which can limit the functionality of that.
